src/Microsoft.Azure.SignalR/DependencyInjectionExtensions.cs

```diff
@@ -36,6 +36,28 @@ public static ISignalRServerBuilder AddAzureSignalR(this ISignalRServerBuilder b
             return builder.AddAzureSignalR<ServiceOptionsSetup>();
         }
 
+        /// <summary>
+        /// Add Azure SignalR with specified connection name, the connection string could be read from ConnectionStrings_{name}, the settings are loaded from Azure:SignalR:{name} section
+        /// </summary>
+        /// <param name="builder">The <see cref="ISignalRServerBuilder"/>.</param>
+        /// <param name="name">The name of the Azure SignalR service that settings and connection strings are read from</param>
+        /// <param name="configure">A callback to configure the <see cref="ServiceOptions"/>.</param>
+        /// <returns>The same instance of the <see cref="ISignalRServerBuilder"/> for chaining.</returns>
+        public static ISignalRServerBuilder AddAzureSignalRWithConnectionName(this ISignalRServerBuilder builder, string name, Action<ServiceOptions> configure = null)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            builder.Services.SetupOptions<ServiceOptions, ServiceOptionsSetup>(s => ActivatorUtilities.CreateInstance<ServiceOptionsSetup>(s, name));
+            if (configure != null)
+            {
+                builder.Services.Configure(configure).PostConfigure<ServiceOptions>(o => o.Validate());
```

Comment:
We should always validate options no matter configure is null or not.

Reply:
Actually I don't think we need to provide the paramter `Action<ServiceOptions>`, because the user can call `builder.Services.Configure(configure)` themselves.

Reply:
https://github.com/dotnet/roslyn-analyzers/blob/main/src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md#rs0026-do-not-add-multiple-public-overloads-with-optional-parameters

Adding multiple public overloads with optional parameter seems not a good practice

Reply:
we don't have method overloads with optional parameter?